scrapper.py

```diff
@@ -25,39 +38,135 @@ class Crawler:
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.is_dir():
+        shutil.rmtree(path)
+    path.mkdir(parents=True, exist_ok=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as config:
+        scrapper_config = json.load(config)
+
+    seed_urls = scrapper_config["seed_urls"]
+    max_articles = scrapper_config["total_articles_to_find_and_parse"]
+
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if not seed_urls:
+        raise IncorrectURLError
+    for seed_url in seed_urls:
+        if URL_PATTERN not in seed_url:
+            raise IncorrectURLError
+
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 300:
+        raise NumberOfArticlesOutOfRangeError
+
+    return seed_urls, max_articles
+
+
+class HTMLParser:
+    """
+    Parser implementation
+    """
+
+    def __init__(self, article_url, article_id):
+        """
+        Init
+        """
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(url=article_url, article_id=article_id)
+
+    def parse(self):
+        """
+        Parses each article
+        """
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+
+        table_bs = article_bs.find('div', class_="fulltext")
+        tag = table_bs.find('a')['href']
+        link = re.sub(r'(?i)view(?=\W)', 'download', tag)
+        pdf_file = PDFRawFile(link, self.article_id)
+
+        pdf_file.download()
+        self.article.text = pdf_file.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+
+        author_bs = article_bs.find('meta', {"name": "DC.Creator.PersonalName"})["content"]
+        if not article_bs:
+            article_bs = 'NOT FOUND'
+        self.article.author = author_bs
+
+        title_bs = article_bs.find('meta', {"name": "description"})['content']
+        self.article.title = title_bs
+
+        date_bs = article_bs.find("meta", {"name": "DC.Date.dateSubmitted"})['content']
+        article_date = datetime.strptime(date_bs, '%Y-%m-%d')
+        self.article.date = article_date
 
 
 if __name__ == '__main__':
-    # YOUR CODE HERE
-    pass
+
+    seed_urls_test, total_articles_test = validate_config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+
+    crawler = Crawler(seed_urls_test, total_articles_test)
+    crawler.find_articles()
+
+    ARTICLE_ID = 0
+    for url in crawler.urls:
+        ARTICLE_ID += 1
```

Comment:
usually counter is incremented at the end of the loop. change accordingly. or maybe you can recall a way of obtaining an index of current element in the iterable object?

Reply:
done